File: 0228-summary-ranges/0228-summary-ranges.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return String[]
     */
    function summaryRanges($nums) {
        if(empty($nums)){
            return [];
        }
        
        $start = $nums[0];
        $nums[] = PHP_INT_MAX;
        $ans = [];
        
        foreach($nums as $k=>$n){
            if(isset($nums[$k - 1]) && ($n - $nums[$k - 1]) > 1){
                $end = $nums[$k - 1];
                if($start == $end){
                    $ans[] = "$start";
                } else {
                    $ans[] = "$start->$end";
                }
                $start = $n;
            }
        }
        
        return $ans;
    }
}

File: 0228-summary-ranges/0228-summary-ranges.go
func summaryRanges(nums []int) []string {
    var ans []string
    numsLen := len(nums)
    
    for i := 0; i < numsLen; i++ {
        start := nums[i]
        
        //this loop help to identify the end of a range of consecutive numbers
        for i < numsLen - 1 && nums[i + 1] == nums[i] + 1 {
            i++
        }
        
        if start == nums[i] {
            ans = append(ans, fmt.Sprintf("%d", nums[i]))
            continue
        }
        
        ans = append(ans, fmt.Sprintf("%d->%d", start, nums[i]))
    }
    
    return ans
}